File: ats-ness-back-prod/src/main/java/nesstechnologies/fr/applicationesnbackend/entities/Segment.java
package nesstechnologies.fr.applicationesnbackend.entities;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
@ToString
public class Segment implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String segmentKey;
    private String segmentValue;
    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "statistiques_profil_id", nullable = false)
    private StatistiquesProfil statistiquesProfil;

}

File: ats-ness-front-prod/src/app/demo/models/Segment.ts
export class Segment {
    id?: number;
    segmentKey: string = '';
    segmentValue: string = '';
    statistiquesProfilId?: number;  // L'ID du StatistiquesProfil auquel ce segment appartient

    constructor(init?: Partial<Segment>) {
        Object.assign(this, init);
    }
}
